File: Sample/Source/Infos/PlayerInfo.py
# -*- encoding: utf-8 -*-

from __future__ import annotations
from Engine.Gameplay.InfoBase import InfoBase
from Source.Configs.GeneralEnum import GeneralDataKey


class PlayerInfo(InfoBase):
    r"""
    \brief Player data layer.

    Defines player identity loaded from GeneralData by ID.
    Independent of Actor; bridged via multiple inheritance on `Player`.
    """

    _infoType: str = GeneralDataKey.Player

File: Sample/Scripts/Source/Infos/PlayerInfo.lua
local Engine = require("Engine")
---@type { GeneralDataKey: Source.Configs.GeneralEnum.GeneralDataKey }
local GeneralEnum = require("Source.Configs.GeneralEnum")

local InfoBase = Engine.InfoBase
local GeneralDataKey = GeneralEnum.GeneralDataKey

local PlayerInfo = {}

PlayerInfo._infoType = GeneralDataKey.Player

return class(PlayerInfo, InfoBase)
